addons/web_editor/static/src/js/backend/html_field.js

```diff
@@ -0,0 +1,554 @@
+import ajax from 'web.ajax';
+import {
+    useBus,
+    useService,
+} from "@web/core/utils/hooks";
+import { getAdjacentPreviousSiblings, getAdjacentNextSiblings } from '@web_editor/js/editor/odoo-editor/src/utils/utils';
+import { toInline } from 'web_editor.convertInline';
+
+const {
+    markup,
+    Component,
+    useRef,
+    useSubEnv,
+    useState,
+    onWillStart,
+    onMounted,
+    onWillUpdateProps,
+    useEffect,
+    onWillUnmount,
+} = owl;
+
+export class HtmlFieldWysiwygAdapterComponent extends ComponentAdapter {
+    setup() {
+        super.setup();
+        useSubEnv(legacyEnv);
+
+        let started = false;
+        onMounted(() => {
+            if (!started) {
+                this.props.startWysiwyg(this.widget);
+                started = true;
+            }
+        });
+    }
+
+    updateWidget(newProps) {
+        if (this.widget.softUpdate) {
+            this.widget.softUpdate = false;
+            return;
+        }
+        this.widget.resetEditor(newProps.widgetArgs[0].value, {
+            collaborationChannel: newProps.widgetArgs[0].collaborationChannel
+        });
+    }
+    renderWidget() {}
+}
+
+export class HtmlField extends Component {
+    setup() {
+        this.readonlyElementRef = useRef("readonlyElement");
+        this.codeViewRef = useRef("codeView");
+        this.iframeRef = useRef("iframe");
+        this.codeViewButtonRef = useRef("codeViewButton");
+
+        this.rpc = useService("rpc");
+
+        this.state = useState({
+            showCodeView: false,
+            iframeVisible: false,
+        });
+
+        useBus(this.env.bus, "RELATIONAL_MODEL:WILL_SAVE_URGENTLY", () => this._commitChangesFromBus({urgent: true}));
+        useBus(this.env.bus, "RELATIONAL_MODEL:NEED_LOCAL_CHANGES", ({detail}) => this._commitChangesFromBus({proms: detail.proms}));
+
+        this._onUpdateIframeId = 'onLoad_' + _.uniqueId('FieldHtml');
+
+        // The wysiwyg.getValue() could return sligthly different value each
+        // call. The commitChanges would in such cases continuously be called as
+        // the `prop.update` will trigger the bus event
+        // `RELATIONAL_MODEL:WILL_SAVE`.
+        this._lastUpdateFromBus = false;
+
+        onWillStart(async () => {
+            this.Wysiwyg = await this._getWysiwygClass();
+            if (this.props.cssReadonlyAssetId) {
+                this.cssReadonlyAsset = await ajax.loadAsset(this.props.cssReadonlyAssetId);
+            }
+            if (this.props.cssEditAssetId || this.props.isInlineStyle) {
+                this.cssEditAsset = await ajax.loadAsset(this.props.cssEditAssetId || 'web_editor.assets_edit_html_field');
+            }
+        });
+        onWillUpdateProps((newProps) => {
+            if (!newProps.readonly && this.state.iframeVisible) {
+                this.state.iframeVisible = false;
+            }
+            if (newProps.value !== this.props.value) {
+                this._resetIframeValue = true;
```

Comment:
Here we check `newProps.value !== this.props.value` then we set `_resetIframeValue` then in `_loadReadonlyIframe` we do `iframeTarget.innerHTML = this.props.value`. Couldn't we avoid using `_resetIframeValue` altogether and check `iframeTarget.innerHTML === this.props.value` in `_loadReadonlyIframe` directly ? What am I missing ?